Source/Foundation/bsfUtility/FileSystem/BsDataStream.cpp

```diff
@@ -94,7 +94,7 @@ namespace bs
 		// Read the entire buffer - ideally in one read, but if the size of
 		// the buffer is unknown, do multiple fixed size reads.
 		size_t bufSize = (mSize > 0 ? mSize : 4096);
-		std::stringstream::char_type* tempBuffer = (std::stringstream::char_type*)bs_alloc((UINT32)bufSize);
+		std::array<std::stringstream::char_type,4096> tempBuffer;
```

Comment:
It might be *slightly* gentler on stack usage to instead use alloca(), but there's probably no major benefit to doing so.

Good find :-)

Reply:
Yeah, 4kb for a function that is not called recursively should not blow up the stack 😄